### PR TITLE
Fix #6: Integrate Google OAuth 2.0 with Ava.

### LINKED ISSUE #6: Email verification Alternate
Due to SendGrid limitations, we need to think of other ways to verify members.

<b>Priority: High</b>

### PR DESCRIPTION
**Pull request by ashutoshc8101**
**Note: Do not merge this into master. It will break Ava. Create a seperate branch.**

Fixes #6

This PR integrates Google OAuth 2.0 for the verification of club members.

The method to verify users remains the same, just the underlying implementation changed.

**Before:**
  The users were asked to provide a verification code sent to them by email.

**Now:**
-   The users will be asked to visit a link to authorize their email ids.
image

-   Once their email ids are authorized, they will be provided an authorization code by Google OAuth 2.0.
image
image

-   The users are required to send the authorization code by DM to Ava instead of the verification code.
image

### CODE REVIEW

File: client_secret.example.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (ashutoshc8101) [inline client_secret.example.json:1]: This is an example file for `client_secret.json`.

**Review by abhishek0220 — COMMENTED** [review]
Seems ok, will test it locally once

File: script/verify.py [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 from pymongo import MongoClient
 
+from oauth.google_auth_service import GoogleAuthService, UserNotValidatedException
```
Comment (abhishek0220) [inline script/verify.py]: typo in import, OAuth should be a package (add init file in that)

File: script/verify.py [hunk 3]
```diff
@@ -81,36 +82,8 @@ async def verify(self, ctx, name : str = "", entry_number : str = ""):
...
-        await ctx.send(f'Name : {name}\nEntry Number : {entry_number}\n{verf_msg}')
-        try:
-            code_got = await self.bot.wait_for('message', check=lambda message: message.author == ctx.author, timeout = 180)
-        except asyncio.TimeoutError:
-            await ctx.send(f"{ctx.author} Verification Time Out")
-            return await logs.print(f'{ctx.author.mention} verification Timeout')
-        if(code_got.content[0] == "."):
-            return
-        if(code_got.content == str(code)):
+        if(self.authorize_user(ctx, logs, name, entry_number)):
```
Comment (abhishek0220) [inline script/verify.py]: add await with == true, so it will not bypass this in exception

File: script/verify.py [hunk 4]
```diff
@@ -159,6 +133,36 @@ async def update_roles(self, ctx):
...
+        verication_message = 'Please visit the following link for verification: {}'
+
+        await ctx.send(f'Name : {name}\nEntry Number : {entry_number}\n' +
+            verication_message.format(auth_url[0]))
+
+        print(ctx.author, 'Verification Started')
+        await logs.print(f'{ctx.author.mention} started Verification')
+
+        try:
+            authorization_code = await self.bot.wait_for(
```
Comment (abhishek0220) [inline script/verify.py]: authorization_code is not the message, use content attribute to get the message
authorization_code  = authorization_code.content

File: script/verify.py (hunk 2 above)
Comment (ashutoshc8101) [reply script/verify.py]: done

File: script/verify.py (hunk 3 above)
Comment (ashutoshc8101) [reply script/verify.py]: done

File: script/verify.py (hunk 4 above)
Comment (ashutoshc8101) [reply script/verify.py]: done

**Review by abhishek0220 — APPROVED** [review]

**Pull request merged**